src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -118,6 +118,13 @@ public AzureAppConfigurationProvider(IConfigurationClientManager configClientMan
 
             if (options.RegisterAllEnabled)
             {
+                if (options.KvCollectionRefreshInterval <= TimeSpan.Zero)
```

Comment:
Just wanted to mention that this shouldn't be possible based on how the refresh interval is set by default if `ConfigureRefresh` and `RegisterAll` were called, but this is fine if we want to just check here for the future